### PR TITLE
add fips compliant packages

### PR DESCRIPTION
**Pull request by rajrohanyadav**
### Summary
Add fips compliant package building.

### Testing
[prerelease workflow](https://github.com/newrelic/nri-redis/actions/runs/11966931908) - The installation test fails from staging as the path is hardcoded (to be fixed as part of [NR-340027](https://new-relic.atlassian.net/browse/NR-340027))

image

[NR-340027]: https://new-relic.atlassian.net/browse/NR-340027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

***NOTE:*** Once [#163 add fips upload schemas to ohi.yml](https://github.com/newrelic/infrastructure-publish-action/pull/163) is merged, the publish action can be upgraded to the new version for this repo. Only after adding these (similar) changes in other integration repos should we update the publish action for them.

### CODE REVIEW

File: .github/workflows/on_prerelease.yaml [hunk 1]
```diff
@@ -9,8 +9,10 @@ on:
 
 jobs:
   pre-release:
-    uses: newrelic/coreint-automation/.github/workflows/reusable_pre_release.yaml@v3
+    uses: newrelic/coreint-automation/.github/workflows/reusable_pre_release.yaml@test-fips-s3-upload-schema
```
Comment (alvarocabanas) [inline .github/workflows/on_prerelease.yaml:12]: I understand this was only used while development and you'll update the v3 once merged, isn't it?

File: .github/workflows/on_prerelease.yaml [hunk 2]
```diff
@@ -9,8 +9,10 @@ on:
...
     with:
       tag: ${{ github.event.release.tag_name }}
+      upload_fips_packages: true
       integration: "redis"
+      dest_prefix: "rohan-fips-test-redis2/"
```
Comment (alvarocabanas) [inline .github/workflows/on_prerelease.yaml:17]: Same, this can't be merged

**Review by alvarocabanas — COMMENTED** [review]
We can't merge this until the actual PRs are merged and reusable_release created

File: .github/workflows/on_prerelease.yaml (hunk 1 above)
Comment (rajrohanyadav) [reply .github/workflows/on_prerelease.yaml:12]: Yes, correct. I'll change it after https://github.com/newrelic/coreint-automation/pull/93 is merged.

Comment (rajrohanyadav) [reply .github/workflows/on_prerelease.yaml]: done

File: .github/workflows/on_prerelease.yaml (hunk 2 above)
Comment (rajrohanyadav) [reply .github/workflows/on_prerelease.yaml]: done

**Pull request merged**